### PR TITLE
Feature/permissions consent

### LINKED ISSUE #256: Increment Consent caused an invalid error
Attempting to call POST /User for the first time, I got the incremental consent dialog, however, I then got this error,

image

I was using EdgeDev. Running the query again worked.

### PR DESCRIPTION
**Pull request by cwahome**
## Overview

- Changes incremental consent implementation to per scope consent
- Removes message directing users to EdgeDev
- Allows login via Redirect / Popup

Fixes #256 

### Demo

image

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* Log in
* Select a query to run
* Open the permissions tab to view the permissions required for the query
* Click the consent button against the permission you need and get the consent required interaction
* Notice that the permission now reads consented

### CODE REVIEW

**Comment by darrelmiller** [comment]
I downloaded this PR, compiled and ran it and it worked perfectly for me.   Nice job @cwahome

File: src/app/views/App.tsx [hunk 1]
```diff
@@ -293,21 +282,7 @@ class App extends Component<IAppProps, IAppState> {
...
-                        <a className={classes.links}
-                          tabIndex={0}
-                          href='https://www.microsoftedgeinsider.com/en-us/download' target='_blank'>
-                          EdgeDev.
-                        </a>
-                      </p>
-                    </MessageBar>
-                  </div>
-                : !showToggle && <><Authentication />
+                {!showToggle && <><Authentication />
```
Comment (andrueastman) [inline src/app/views/App.tsx]: 👍

**Comment by andrueastman** [comment]
This works really well for me on Edge and Chrome. 
Maybe in future we could add a message to direct the user to the Permissions tab when a 403 occurs to be more clear.

**Comment by cwahome** [comment]
> I downloaded this PR, compiled and ran it and it worked perfectly for me. Nice job @cwahome

Awesome :-) Could not have done it without @irvinesunday

**Pull request merged**